### PR TITLE
Add types to the plugin config definitions

### PR DESCRIPTION
**Pull request by Tatskaari**

### CODE REVIEW

File: docs/config.html [hunk 1]
```diff
@@ -448,6 +448,17 @@ <h3 class="mt1 f6 lh-title" id="pluginconfig.repeatable">
         </p>
       </div>
     </li>
+    <li>
+      <div>
+        <h3 class="mt1 f6 lh-title" id="pluginconfig.boolean">
+          Boolean <span class="normal">(bool)</span>
```
Comment (peterebden) [inline docs/config.html]: This is out of line with below, no?

File: src/parse/asp/objects.go [hunk 2]
```diff
@@ -967,6 +968,37 @@ func loadPluginConfig(state *core.BuildState, c pyDict) {
 	c[strings.ToUpper(pluginName)] = pluginNamespace
 }
 
+func toPyObject(key, val, toType string) pyObject {
+	if toType == "" || toType == "str" {
+		return pyString(val)
+	}
+
+	if val == "" {
+		return pyNone{}
```
Comment (peterebden) [inline src/parse/asp/objects.go:977]: Should we do this? Or should they become the zero value (0 / False)?

File: src/parse/asp/objects.go [hunk 3]
```diff
@@ -967,6 +968,37 @@ func loadPluginConfig(state *core.BuildState, c pyDict) {
...
+	}
+
+	if toType == "bool" {
+		if val == "true" || val == "True" {
```
Comment (peterebden) [inline src/parse/asp/objects.go:981]: The config files support yes/no and on/off as well; should we include that here too?

File: src/parse/asp/objects.go (hunk 2 above)
Comment (Tatskaari) [reply src/parse/asp/objects.go:977]: This is the difference between an optional boolean value that hasn't been set, and one that's explicitly been set to False. Perhaps we don't need to model that but it seems more correct.

File: src/parse/asp/objects.go (hunk 3 above)
Comment (Tatskaari) [reply src/parse/asp/objects.go]: Good point. Done.

File: docs/config.html (hunk 1 above)
Comment (Tatskaari) [reply docs/config.html]: you were too fest

File: src/parse/asp/objects.go (hunk 2 above)
Comment (Tatskaari) [reply src/parse/asp/objects.go]: Yeah okay I'll change this.

**Review by tiagovtristao — APPROVED** [review]
Lgtm. As discussed offline, I agree with the zero-value consistency that already exists for configuration. I just wonder how the `Optional` field plays with it and whether it might cause some confusion.

**Comment by Tatskaari** [comment]
> Lgtm. As discussed offline, I agree with the zero-value consistency that already exists for configuration. I just wonder how the `Optional` field plays with it and whether it might cause some confusion.

I guess the Optional field now means that the value has to be non-zero i.e. not 0 or "". Not sure it makes sense for bools.

**Pull request merged**